app/src/main/java/space/taran/arknavigator/utils/FileUtils.kt

```diff
@@ -18,38 +32,181 @@ enum class Sorting {
     DEFAULT, NAME, SIZE, LAST_MODIFIED, TYPE
 }
 
-enum class FileActionType{
+enum class FileActionType {
     EDIT_AS_OPEN, EDIT_AND_OPEN, OPEN_ONLY, OPEN_ONLY_DETACH_PROCESS
 }
 
+enum class FileType {
+    IMAGE, VIDEO, GIF, PDF, UNDEFINED
+}
+
+const val PDF_PREVIEW_FOLDER_NAME = "pdf_preview"
+
+class PreviewFile(
+    val fileType: FileType? = null,
+    val file: Path? = null,
+    val fileExtension: String? = null,
+    val extraInfo: MutableMap<Preview.ExtraInfoTag, String>? = null
+) {
+    companion object PreviewCompanion {
+        fun createPair(file: Path): PreviewFile {
+            return when {
+                isImage(file) -> {
+                    PreviewFile(
+                        FileType.IMAGE,
+                        file,
+                        fileExtension = extensionWithoutDot(file)
+                    )
+                }
+                isVideo(file) -> {
+                    PreviewFile(
+                        FileType.VIDEO,
+                        file,
+                        fileExtension = extensionWithoutDot(file),
+                        extraInfo = getVideoInfo(file)
+                    )
+                }
+                isPDF(file) -> getPdfPreview(file)
+                isFormat(file, ".gif") ->
+                    PreviewFile(
+                        FileType.GIF,
+                        file,
+                        fileExtension = extensionWithoutDot(file)
+                    )
+                else -> PreviewFile(fileExtension = extensionWithoutDot(file))
+            }
+        }
+
+        private fun getPdfPreview(file: Path): PreviewFile {
+            val id = computeId(file)
+            val savedPreviews = getSavedPdfPreviews()
+            return if (savedPreviews?.contains(id) == true) {
```

Comment:
Not sure here we need to recalculate the `id`. But with the current architecture it doesn't seem like it can be done otherwise.

Reply:
This is a good point. It's possible to pass the id beforehand, but:
Out of the 3 places, where the method `Preview.provide(path: Path)` is called, the id is currently available in `GalleryPresenter` and `ResourcesGridPresenter`. However in `FoldersWalker` it's not directly available yet. I will remake it so that when we have the id available, we won't have to compute it.